### PR TITLE
Allow partial span validation on parent on only one date field

### PR DESCRIPTION
**Pull request by danielaSerban**
- [Enrollment requirement](https://annkissam-team.monday.com/boards/332596247/pulses/883808047)

### CODE REVIEW

File: lib/acts_as_span/within_parent_date_span_validator.rb [hunk 1]
```diff
@@ -15,14 +15,24 @@ def outside_of_parent_date_span?(record, parent_sym)
...
       return false if parent.nil?
 
-        child_record_without_start_date(record, parent) ||
-        child_record_without_end_date(record, parent) ||
-        child_record_started_before_parent_record(record, parent) ||
-          child_record_ended_after_parent_record(record, parent)
+      start_date_invalid = options[:skip_start_date_validation] ? false : start_date_out_of_span?(record, parent)
+      end_date_invalid = options[:skip_end_date_validation] ? false : end_date_out_of_span?(record, parent)
+
+      start_date_invalid || end_date_invalid
```
Comment (izzergh) [inline lib/acts_as_span/within_parent_date_span_validator.rb:21]: I would split this up into full `if`s; the lines are starting to get unwieldy

**Review by izzergh — CHANGES_REQUESTED** [review]
What you are testing is that you can remove the overlap validation entirely for a model - could you make a model for skipping start and another for skipping end, and test that the overlap validator still cares about the other one?

**Comment by danielaSerban** [comment]
> What you are testing is that you can remove the overlap validation entirely for a model - could you make a model for skipping start and another for skipping end, and test that the overlap validator still cares about the other one?

Ayee, you're right. I'm gonna add some more tests.

**Comment by izzergh** [comment]
Oof and it looks like CI broke...

**Comment by izzergh** [comment]
@danielaSerban if you merge #25 into this, Travis will run properly (but if you merge this before #25 gets merged, you should bump the version since it locks Rails)

**Comment by izzergh** [comment]
👀

**Pull request merged**